packages/code-du-travail-frontend/src/search/Search.js

```diff
@@ -208,104 +212,98 @@ class Search extends React.Component {
       });
     }
   };
-  setResults = results => {
-    if (this.props.onResults && results && results.hits.total > 0) {
-      this.props.onResults(results.hits.hits);
-    }
+
+  onSelect = suggestion => {
+    const { query } = this.state;
+    const route = getRouteBySource(suggestion._source.source);
+    const anchor = suggestion._source.anchor
+      ? suggestion._source.anchor.slice(1)
+      : "";
+
+    Router.pushRoute(
+      route,
+      { q: query, search: 0, slug: suggestion._source.slug },
+      { hash: anchor }
+    );
+  };
+
+  onClear = () => {
+    this.setState({ suggestions: [] });
+  };
+
+  onSearch = ({ value }) => {
+    const { excludeSources } = this.state;
+    fetchResultsSuggest(value, excludeSources).then(results => {
+      this.setState({ suggestions: results.hits.hits });
+      this.props.onResults(results);
```

Comment:
maybe nest the `onResults` call in the `setState` callback ?

Reply:
I see htat pattern already but I don't know why ? this defer the callback ?

Reply:
yes, this ensure the order is respected by triggering the callback only when state is effectively updated.